src/main/java/com/exasol/releasedroid/adapter/github/GitHubAPIAdapter.java

```diff
@@ -29,27 +20,31 @@
  */
 public class GitHubAPIAdapter implements GitHubGateway {
     private static final Logger LOGGER = Logger.getLogger(GitHubAPIAdapter.class.getName());
-    private static final String GITHUB_API_ENTRY_URL = "https://api.github.com/repos/";
     private static final String PREPARE_ORIGINAL_CHECKSUM_WORKFLOW = "release_droid_prepare_original_checksum.yml";
     private static final String GITHUB_RELEASE_WORKFLOW = "release_droid_upload_github_release_assets.yml";
     private static final String PRINT_QUICK_CHECKSUM_WORKFLOW = "release_droid_print_quick_checksum.yml";
-    private final Map<String, GHRepository> repositories;
-    private final User gitHubUser;
+    private final Map<String, GHRepository> repositories = new HashMap<>();
+    private final GitHub gitHub;
 
     /**
      * Create a new instance of {@link GitHubAPIAdapter}.
      *
-     * @param gitHubUser instance of {@link User}
+     * @param gitHub instance of {@link GitHub}
      */
-    public GitHubAPIAdapter(final User gitHubUser) {
-        this.gitHubUser = gitHubUser;
-        this.repositories = new HashMap<>();
+    public GitHubAPIAdapter(final GitHub gitHub) {
+        this.gitHub = gitHub;
     }
 
-    private GHRepository createGHRepository(final String repositoryName, final User user) throws GitHubException {
+    private GHRepository getRepository(final String repositoryName) throws GitHubException {
+        if (!this.repositories.containsKey(repositoryName)) {
+            this.repositories.put(repositoryName, this.createGHRepository(repositoryName));
+        }
+        return this.repositories.get(repositoryName);
+    }
```

Comment:
I think you can use `computeIfAbsent` here

Reply:
This won't here because I want to throw an exception which comes from createGHRepository() and lambda can't handle it, It asks to surround it with try-catch which I don't want